DetectorDescription/RecoGeometry/src/DTGeometryBuilder.cc

```diff
@@ -0,0 +1,282 @@
+// Class:      DTGeometryESProducer
+// 
+/**\class DTGeometryESProducer
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Ianna Osborne
+//         Created:  Wed, 16 Jan 2019 10:19:37 GMT
+//
+//
+#include "CondFormats/GeometryObjects/interface/RecoIdealGeometry.h"
+#include "CondFormats/Alignment/interface/DetectorGlobalPosition.h"
+#include "CondFormats/Alignment/interface/AlignmentErrorsExtended.h"
+#include "CondFormats/AlignmentRecord/interface/GlobalPositionRcd.h"
+#include "CondFormats/AlignmentRecord/interface/DTAlignmentRcd.h"
+#include "CondFormats/AlignmentRecord/interface/DTAlignmentErrorRcd.h"
+#include "CondFormats/AlignmentRecord/interface/DTAlignmentErrorExtendedRcd.h"
+#include "Geometry/CommonTopologies/interface/GeometryAligner.h"
+#include "Geometry/Records/interface/DTRecoGeometryRcd.h"
+#include "DataFormats/GeometrySurface/interface/Plane.h"
+#include "DataFormats/GeometrySurface/interface/Bounds.h"
+#include "DataFormats/GeometrySurface/interface/RectangularPlaneBounds.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/ESProductHost.h"
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/ReusableObjectHolder.h"
+#include "DetectorDescription/DDCMS/interface/MuonNumbering.h"
+#include "DetectorDescription/DDCMS/interface/MuonNumberingRcd.h"
+#include "DetectorDescription/DDCMS/interface/MuonGeometryRcd.h"
+#include "DetectorDescription/DDCMS/interface/DDSpecParRegistryRcd.h"
+#include "DetectorDescription/DDCMS/interface/DDSpecParRegistry.h"
+#include "DetectorDescription/DDCMS/interface/DetectorDescriptionRcd.h"
+#include "DetectorDescription/DDCMS/interface/DDDetector.h"
+#include "DetectorDescription/DDCMS/interface/DDFilteredView.h"
+#include "DetectorDescription/DDCMS/interface/DDUnits.h"
+#include "Geometry/DTGeometry/interface/DTGeometry.h"
+#include "DetectorDescription/RecoGeometry/interface/DTNumberingScheme.h"
+#include "DetectorDescription/RecoGeometry/interface/DTGeometryBuilder.h"
+#include "DD4hep/Detector.h"
+#include "DD4hep/DetectorTools.h"
+#include "DD4hep/VolumeProcessor.h"
+#include "TGeoManager.h"
+#include "TClass.h"
+
+#include <memory>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <string_view>
+#include <regex>
+
+using namespace edm;
+using namespace std;
+using namespace cms;
+
+namespace {
+  
+  string_view noNamespace(string_view input) {
+    string_view v = input;
+    auto first = v.find_first_of(":");
+    v.remove_prefix(min(first+1, v.size()));
+    return v;
+  }
+}
+
+void
+DTGeometryBuilder::buildGeometry(DDFilteredView& fview, dd4hep::Volume top,
```

Comment:
What specifically would you like to see improved?

Reply:
The builder takes 9.2 sec vs 1.2 original DD. I'm checking with TGeo experts if other search algorithms could be applied here. Something like a red-black tree to avoid a regex comparison on the nodes which are not selected.